providers/core/core.rtcdatachannel.js

```diff
@@ -124,6 +125,11 @@ RTCDataChannelAdapter.prototype.onclose = function (event) {
 RTCDataChannelAdapter.prototype.onmessage = function (event) {
   if (typeof event.data === 'string') {
     this.dispatchEvent('onmessage', {text: event.data});
+  } else if (this.channel.binaryType === 'arraybuffer' &&
```

Comment:
That's a bug about ArrayBuffer conversion not working as expected in addons. This issue is that the WebRTC object returned is owned by the hiddenWindow rather than the addon, and needs to be cloned into our context to establish ownership.

The intention may well be that the clone should be explicit, since it allows the module to support the mode of getting a reference to the object and then passing that reference to a consumer without forcing an additional memory copy of the contained data.